2017-2018/botCode/pacbotSimulatorModule.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+
+import os
+import robomodules as rm
+from messages import MsgType, message_buffers, PacmanState
+
+ADDRESS = os.environ.get("LOCAL_ADDRESS","localhost")
+PORT = os.environ.get("LOCAL_PORT", 11295)
+
+
+class PacbotSimulatorModule(rm.ProtoModule):
+    def __init__(self, addr, port):
+        self.subscriptions = [MsgType.PACMAN_COMMAND]
+        super().__init__(addr, port, message_buffers, MsgType, self.subscriptions)  # Frequency is 0 by default
+        self.current_location = (14, 7)  # I can't figure out how to import the starting location of pacbot
+
+    def msg_received(self, msg, msg_type):
```

Comment:
The idea is that the high level code is going to continuously calculate the best move possible as fast as possible. Probably at around 60Hz. As such, with the current setup, this module is going to receive 60 commands a second and pacbot is going to FLY through the maze.

What you should do instead:
- have a variable that keeps track of the latest command received.
- Whenever you receive a command message, update that variable
- Set some speed frequency (the frequency should match the frequency with which we expect the bot to actually be able to execute moves. Eg. If we think the bot can move 1 square in 2 seconds, then the frequency should be 0.5 (Hz)).
- In the tick function, look at the most recent command that you've received and execute that command. In this case executing just means updating the location and sending the message.

Reply:
👍 , what is our actual expected speed of the bot? Did we say 2 squares per second on Saturday?